### ISSUE TITLE
Impossible to install with yarn and node 8

### ISSUE THREAD

**Issue by himdel** [patternfly/patternfly-timeline#52]
```
$ node --version
v8.11.2
$ yarn --version
1.9.4

$ mkdir magic
$ cd magic
$ yarn init .   #just press enter until done
$ yarn add patternfly-timeline
```

Errors with:

```
yarn add v1.9.4
info No lockfile found.
[1/4] Resolving packages...
warning patternfly-timeline > patternfly > bootstrap-treeview > ejs@2.2.4: Critical security bugs fixed in 2.5.5
warning patternfly-timeline > patternfly > bootstrap-treeview > express > connect@2.12.0: connect 2.x series is deprecated
warning patternfly-timeline > patternfly > bootstrap-treeview > phantomjs > request > tough-cookie@2.2.2: ReDoS vulnerability parsing Set-Cookie https://nodesecurity.io/advisories/130
warning patternfly-timeline > patternfly > bootstrap-treeview > phantomjs > request > node-uuid@1.4.8: Use uuid module instead
warning patternfly-timeline > patternfly > bootstrap-treeview > phantomjs > request > hawk > hoek@2.16.3: The major version is no longer supported. Please update to 4.x or newer
warning patternfly-timeline > patternfly > bootstrap-treeview > phantomjs > request > hawk > boom > hoek@2.16.3: The major version is no longer supported. Please update to 4.x or newer
warning patternfly-timeline > patternfly > bootstrap-treeview > phantomjs > request > hawk > sntp > hoek@2.16.3: The major version is no longer supported. Please update to 4.x or newer
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
[1/3] ⠁ contextify
[2/3] ⠁ phantomjs
warning Error running install script for optional dependency: "/home/himdel/magic/node_modules/phantomjs: Command failed.
Exit code: 1
Command: node install.js
Arguments: 
Directory: /home/himdel/magic/node_modules/phantomjs
Output:
PhantomJS not found on PATH
Phantom installation failed TypeError: Path must be a string. Received undefined
    at assertPath (path.js:28:11)
    at Object.join (path.js:1236:7)
    at findSuitableTempDirectory (/home/himdel/magic/node_modules/phantomjs/install.js:127:30)
    at /home/himdel/magic/node_modules/phantomjs/install.js:476:19
    at nextTickCallback (/home/himdel/magic/node_modules/kew/kew.js:47:28)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9) TypeError: Path must be a string. Received undefined
    at assertPath (path.js:28:11)
    at Object.join (path.js:1236:7)
    at findSuitableTempDirectory (/home/himdel/magic/node_modules/phantomjs/install.js:127:30)
    at /home/himdel/magic/node_modules/phantomjs/install.js:476:19
    at nextTickCallback (/home/himdel/magic/node_modules/kew/kew.js:47:28)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
[1/3] ⠐ contextify
[3/3] ⠐ bootstrap-treeview
error /home/himdel/magic/node_modules/contextify: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /home/himdel/magic/node_modules/contextify
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@8.11.2 | linux | x64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/himdel/magic/node_modules/contextify/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/himdel/.node-gyp/8.11.2/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/himdel/.node-gyp/8.11.2',
gyp info spawn args   '-Dnode_gyp_dir=/usr/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/himdel/.node-gyp/8.11.2/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/himdel/magic/node_modules/contextify',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/himdel/magic/node_modules/contextify/build'
  CXX(target) Release/obj.target/contextify/src/contextify.o
In file included from /home/himdel/.node-gyp/8.11.2/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/himdel/.node-gyp/8.11.2/include/node/v8.h: In static member function ‘static void v8::V8::SetAllowCodeGenerationFromStringsCallback(v8::DeprecatedAllowCodeGenerationFromStringsCallback)’:
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:10230:19: warning: cast between incompatible function types from ‘v8::DeprecatedAllowCodeGenerationFromStringsCallback’ {aka ‘bool (*)(v8::Local<v8::Context>)’} to ‘v8::FreshNewAllowCodeGenerationFromStringsCallback’ {aka ‘bool (*)(v8::Local<v8::Context>, v8::Local<v8::String>)’} [-Wcast-function-type]
           callback));
                   ^
/home/himdel/.node-gyp/8.11.2/include/node/v8.h: In static member function ‘static void v8::V8::RemoveGCPrologueCallback(v8::GCCallback)’:
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:10275:53: warning: cast between incompatible function types from ‘v8::GCCallback’ {aka ‘void (*)(v8::GCType, v8::GCCallbackFlags)’} to ‘v8::Isolate::GCCallback’ {aka ‘void (*)(v8::Isolate*, v8::GCType, v8::GCCallbackFlags)’} [-Wcast-function-type]
       reinterpret_cast<Isolate::GCCallback>(callback));
                                                     ^
/home/himdel/.node-gyp/8.11.2/include/node/v8.h: In static member function ‘static void v8::V8::RemoveGCEpilogueCallback(v8::GCCallback)’:
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:10282:53: warning: cast between incompatible function types from ‘v8::GCCallback’ {aka ‘void (*)(v8::GCType, v8::GCCallbackFlags)’} to ‘v8::Isolate::GCCallback’ {aka ‘void (*)(v8::Isolate*, v8::GCType, v8::GCCallbackFlags)’} [-Wcast-function-type]
       reinterpret_cast<Isolate::GCCallback>(callback));
                                                     ^
In file included from ../src/contextify.cc:3:
../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../nan/nan.h:2200:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/contextify.cc: In static member function ‘static v8::Local<v8::Context> ContextWrap::createV8Context(v8::Local<v8::Object>)’:
../src/contextify.cc:131:68: warning: ‘v8::Local<v8::Object> v8::Function::NewInstance() const’ is deprecated: Use maybe version [-Wdeprecated-declarations]
         Local<Object> wrapper = Nan::New(constructor)->NewInstance();
                                                                    ^
In file included from /home/himdel/.node-gyp/8.11.2/include/node/v8.h:26,
                 from /home/himdel/.node-gyp/8.11.2/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:3851:52: note: declared here
   V8_DEPRECATED("Use maybe version", Local<Object> NewInstance() const);
                                                    ^~~~~~~~~~~
/home/himdel/.node-gyp/8.11.2/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/contextify.cc:150:16: error: ‘class v8::ObjectTemplate’ has no member named ‘SetAccessCheckCallbacks’; did you mean ‘SetAccessCheckCallback’?
         otmpl->SetAccessCheckCallbacks(GlobalPropertyNamedAccessCheck,
                ^~~~~~~~~~~~~~~~~~~~~~~
                SetAccessCheckCallback
../src/contextify.cc: In static member function ‘static void ContextWrap::GlobalPropertyGetter(v8::Local<v8::String>, const Nan::PropertyCallbackInfo<v8::Value>&)’:
../src/contextify.cc:182:80: warning: ‘v8::Local<v8::Value> v8::Object::GetRealNamedProperty(v8::Local<v8::String>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
         Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
                                                                                ^
In file included from /home/himdel/.node-gyp/8.11.2/include/node/v8.h:26,
                 from /home/himdel/.node-gyp/8.11.2/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:3404:30: note: declared here
                 Local<Value> GetRealNamedProperty(Local<String> key));
                              ^~~~~~~~~~~~~~~~~~~~
/home/himdel/.node-gyp/8.11.2/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/contextify.cc: In static member function ‘static void ContextWrap::GlobalPropertyQuery(v8::Local<v8::String>, const Nan::PropertyCallbackInfo<v8::Integer>&)’:
../src/contextify.cc:209:67: warning: ‘v8::Local<v8::Value> v8::Object::GetRealNamedProperty(v8::Local<v8::String>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
         if (!Nan::New(ctx->sandbox)->GetRealNamedProperty(property).IsEmpty() ||
                                                                   ^
In file included from /home/himdel/.node-gyp/8.11.2/include/node/v8.h:26,
                 from /home/himdel/.node-gyp/8.11.2/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:3404:30: note: declared here
                 Local<Value> GetRealNamedProperty(Local<String> key));
                              ^~~~~~~~~~~~~~~~~~~~
/home/himdel/.node-gyp/8.11.2/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/contextify.cc:210:71: warning: ‘v8::Local<v8::Value> v8::Object::GetRealNamedProperty(v8::Local<v8::String>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
             !Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property).IsEmpty()) {
                                                                       ^
In file included from /home/himdel/.node-gyp/8.11.2/include/node/v8.h:26,
                 from /home/himdel/.node-gyp/8.11.2/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:3404:30: note: declared here
                 Local<Value> GetRealNamedProperty(Local<String> key));
                              ^~~~~~~~~~~~~~~~~~~~
/home/himdel/.node-gyp/8.11.2/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/contextify.cc:1:
../src/contextify.cc: At global scope:
/home/himdel/.node-gyp/8.11.2/include/node/node.h:474:43: warning: cast between incompatible function types from ‘void (*)(v8::Local<v8::Object>)’ to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
       (node::addon_register_func) (regfunc),                          \
                                           ^
/home/himdel/.node-gyp/8.11.2/include/node/node.h:505:3: note: in expansion of macro ‘NODE_MODULE_X’
   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
   ^~~~~~~~~~~~~
../src/contextify.cc:361:5: note: in expansion of macro ‘NODE_MODULE’
     NODE_MODULE(contextify, init)
     ^~~~~~~~~~~
In file included from /home/himdel/.node-gyp/8.11.2/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/himdel/.node-gyp/8.11.2/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]’:
/home/himdel/.node-gyp/8.11.2/include/node/node_object_wrap.h:85:78:   required from here
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:9253:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<node::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/himdel/.node-gyp/8.11.2/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]’:
../../nan/nan_object_wrap.h:66:61:   required from here
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:9253:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
In file included from ../../nan/nan.h:2686,
                 from ../src/contextify.cc:3:
../../nan/nan_weak.h: In instantiation of ‘Nan::WeakCallbackInfo<P>::WeakCallbackInfo(Nan::Persistent<v8::Value>*, Nan::WeakCallbackInfo<P>::Callback, void*, void*, void*) [with T = ContextWrap; Nan::WeakCallbackInfo<P>::Callback = void (*)(const Nan::WeakCallbackInfo<ContextWrap>&)]’:
../../nan/nan_weak.h:264:12:   required from ‘void Nan::Persistent<T, M>::SetWeak(P*, typename Nan::WeakCallbackInfo<P>::Callback, Nan::WeakCallbackType) [with P = ContextWrap; T = v8::Object; M = v8::NonCopyablePersistentTraits<v8::Object>; typename Nan::WeakCallbackInfo<P>::Callback = void (*)(const Nan::WeakCallbackInfo<ContextWrap>&); Nan::WeakCallbackType = v8::WeakCallbackType]’
../src/contextify.cc:136:132:   required from here
../../nan/nan_weak.h:58:16: warning: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘class v8::Persistent<v8::Value>’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Wclass-memaccess]
     std::memcpy(&persistent_, persistent, sizeof (v8::Persistent<v8::Value>));
     ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/himdel/.node-gyp/8.11.2/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:677:35: note: ‘class v8::Persistent<v8::Value>’ declared here
 template <class T, class M> class Persistent : public PersistentBase<T> {
                                   ^~~~~~~~~~
/home/himdel/.node-gyp/8.11.2/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::WeakCallbackInfo<ContextWrap>; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::WeakCallbackInfo<ContextWrap> >&)]’:
../../nan/nan_weak.h:268:35:   required from ‘void Nan::Persistent<T, M>::SetWeak(P*, typename Nan::WeakCallbackInfo<P>::Callback, Nan::WeakCallbackType) [with P = ContextWrap; T = v8::Object; M = v8::NonCopyablePersistentTraits<v8::Object>; typename Nan::WeakCallbackInfo<P>::Callback = void (*)(const Nan::WeakCallbackInfo<ContextWrap>&); Nan::WeakCallbackType = v8::WeakCallbackType]’
../src/contextify.cc:136:132:   required from here
/home/himdel/.node-gyp/8.11.2/include/node/v8.h:9253:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::WeakCallbackInfo<ContextWrap> >::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::WeakCallbackInfo<ContextWrap> >&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [contextify.target.mk:101: Release/obj.target/contextify/src/contextify.o] Error 1
make: Leaving directory '/home/himdel/magic/node_modules/contextify/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Linux 4.18.0-1-amd64
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/himdel/magic/node_modules/contextify
gyp ERR! node -v v8.11.2
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
```

Not sure about the cause, as neither phantomjs or contextify are actually dependencies of patternfly-timeline.

~~But almost sure this is happening because the `dependencies` field in package.json  should really be `peerDependencies` instead, otherwise `patternfly-timeline` would need to actually expose d3 and patternfly.~~

**Comment by himdel** [comment]
Actually... looks like this is fixed in master, but not in the latest release...

```
yarn add https://github.com/patternfly/patternfly-timeline#1.0.7
```

fails with the exact same problem.

But..

```
yarn add https://github.com/patternfly/patternfly-timeline#master
```

works fine :).

**Comment by himdel** [comment]
Fixed by https://github.com/patternfly/patternfly-timeline/pull/42

Leaving open as this fix from a year ago was never released....

**Comment by himdel** [comment]
So.. I guess the project is dead..

Fixed in @pf3/timeline 1.0.8

(from https://github.com/himdel/patternfly-timeline)
